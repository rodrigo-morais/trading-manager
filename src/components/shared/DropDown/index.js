import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'


const Container = styled.div`
  display: inline-grid;
  grid-template-columns: 40% 60%;
  min-width: 140px;
`
const Label = styled.label`
  font-size: 18px;
  font-weight: bold;
  grid-column-start: 1;
  grid-column-end: 1;
  margin-right: 8px;
`

const Select = styled.select`
  grid-column-start: 2;
  grid-column-end: 2;
  min-height: 96%;
  margin-top: 2%;
  color: #000000;
`

const addColon = (label) => label.slice(-1) === ':' ? label : `${label}:`

const DropDown = ({ label, options, disabled }) => (
  <Container>
    <Label htmlFor={label}>{addColon(label)}</Label>
    <Select name={label} id={label} disabled={disabled}>
      {options.map((option) => (<option key={option} value={option}>{option}</option>))}
    </Select>
  </Container>
)

DropDown.defaultProps = {
  disabled: false,
}

DropDown.propTypes = {
	label: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  disabled: PropTypes.bool,
}

export default DropDown
