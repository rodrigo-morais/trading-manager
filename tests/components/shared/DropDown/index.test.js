import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import DropDown from '../../../../src/components/shared/DropDown'

const onChange = jest.fn()

describe('DropDown', () => {
  describe('Rendering', () => {
    describe('enabled DropDown', () => {
      const wrapper = (
          <DropDown
            value=""
            label="Ativo"
            options={['WINM20', 'WDOM20']}
            onChange={onChange}
          />
      )

      it('renders according to design', () => {
        const component = renderer.create(wrapper)
        const tree = component.toJSON()

        expect(tree).toMatchSnapshot()
      })

      it('has 2 items', () => {
        const component = mount(wrapper)
        expect(component.find('select').find('option').length).toEqual(2)
      })

      it('has a value as "WINM20"', () => {
        const component = mount(wrapper)
        expect(component.find('select').find('option').first().text()).toEqual('WINM20')
      })

      it('has a label with "Ativo" as text', () => {
        const component = mount(wrapper)
        expect(component.find('label').text()).toEqual('Ativo:')
      })

      it('executes onChange callback', () => {
        const component = mount(wrapper)

        component.find('select').simulate('change', { target: { value: 'WDON20' } })
        expect(onChange).toBeCalled();
      })
    })

    describe('enabled DropDown with options as objects', () => {
      const wrapper = (
          <DropDown
            value=""
            label="Sistema"
            options={[{ id: 1, text: 'system 1' }, { id: 2, text: 'system 2' }]}
            onChange={onChange}
          />
      )

      it('renders according to design', () => {
        const component = renderer.create(wrapper)
        const tree = component.toJSON()

        expect(tree).toMatchSnapshot()
      })

      it('has 2 items', () => {
        const component = mount(wrapper)
        expect(component.find('select').find('option').length).toEqual(2)
      })

      it('has the first object with a text as "system 1"', () => {
        const component = mount(wrapper)
        expect(component.find('select').find('option').first().text()).toEqual('system 1')
      })

      it('has a label with "Sistema" as text', () => {
        const component = mount(wrapper)
        expect(component.find('label').text()).toEqual('Sistema:')
      })

      it('executes onChange callback', () => {
        const component = mount(wrapper)

        component.find('select').simulate('change', { target: { value: 'WDON20' } })
        expect(onChange).toBeCalled();
      })
    })

    describe('disabled DropDown', () => {
      const wrapper = (
          <DropDown
            value=""
            label="Ativo"
            options={['WINM20', 'WDOM20']}
            disabled
            onChange={() => {}}
          />
      )

      it('renders according to design', () => {
        const component = renderer.create(wrapper)
        const tree = component.toJSON()

        expect(tree).toMatchSnapshot()
      })

      it('has a drop down as a label', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(1).text()).toEqual('')
      })

      it('has a label with "Ativo" as text', () => {
        const component = mount(wrapper)
        expect(component.find('label').at(0).text()).toEqual('Ativo:')
      })
    })
  })
})
