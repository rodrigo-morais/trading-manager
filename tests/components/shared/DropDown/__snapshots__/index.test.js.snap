// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown Rendering disabled DropDown renders according to design 1`] = `
<div
  className="sc-bdVaJa dzArWz"
>
  <label
    className="sc-bwzfXH jnEWcg"
    htmlFor="Ativo"
  >
    Ativo:
  </label>
  <select
    className="sc-htpNat dlmdsH"
    disabled={true}
    id="Ativo"
    name="Ativo"
  >
    <option
      value="WINM20"
    >
      WINM20
    </option>
    <option
      value="WDOM20"
    >
      WDOM20
    </option>
  </select>
</div>
`;

exports[`DropDown Rendering enabled DropDown renders according to design 1`] = `
<div
  className="sc-bdVaJa dzArWz"
>
  <label
    className="sc-bwzfXH jnEWcg"
    htmlFor="Ativo"
  >
    Ativo:
  </label>
  <select
    className="sc-htpNat dlmdsH"
    disabled={false}
    id="Ativo"
    name="Ativo"
  >
    <option
      value="WINM20"
    >
      WINM20
    </option>
    <option
      value="WDOM20"
    >
      WDOM20
    </option>
  </select>
</div>
`;
